File: haskell/iniciante/1007.hs
-- title: Diferença
-- number of the problem: 1007
-- date: 20/09/20
-- author: ltsdw

myProduct :: (Int, Int, Int, Int) -> (Int, Int)
myProduct (a, b, c, d) = (a * b, c * d)

myDifference :: (Int, Int) -> Int
myDifference (prod_a, prod_b) = prod_a - prod_b

main :: IO ()
main = do
    a' <- getLine
    b' <- getLine
    c' <- getLine
    d' <- getLine

    let a = read a' :: Int
    let b = read b' :: Int
    let c = read c' :: Int
    let d = read d' :: Int

    let prod = myProduct (a, b, c, d)
    let diff = myDifference prod

    putStrLn $ "DIFERENCA = " ++ show diff

File: c++/iniciante/src/1007.cpp
/* title: Diferença
* number of the problem: 1007
* date: 29/08/21
* author: ltsdw */

#include <iostream>
#include <array>

using std::array;

void init_array(array<long, 4>& nums)
{
    using std::cin;

    for (long& num : nums) cin >> num;
}

long prod_difference(const array<long, 4>& nums)
{
    return ((nums[0] * nums[1]) - (nums[2] * nums[3]));
}

int main(int argc, const char* argv[])
{
    using std::cout;

    array<long, 4> nums{};
    init_array(nums);

    cout << "DIFERENCA = " << prod_difference(nums) << '\n';

    return EXIT_SUCCESS;
}
